appholder/src/main/java/com/ul/ims/gmdl/appholder/document/DocumentManager.kt

```diff
@@ -0,0 +1,159 @@
+package com.ul.ims.gmdl.appholder.document
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.security.keystore.KeyProperties
+import androidx.security.identity.Helpers
+import androidx.security.identity.IdentityCredentialException
+import androidx.security.identity.IdentityCredentialStore
+import com.ul.ims.gmdl.appholder.R
+import com.ul.ims.gmdl.appholder.util.DocumentData
+import com.ul.ims.gmdl.appholder.util.DocumentData.DUMMY_CREDENTIAL_NAME
+import com.ul.ims.gmdl.appholder.util.DocumentData.MDL_DOCTYPE
+import org.bouncycastle.asn1.x500.X500Name
+import org.bouncycastle.cert.jcajce.JcaX509v3CertificateBuilder
+import org.bouncycastle.operator.ContentSigner
+import org.bouncycastle.operator.OperatorCreationException
+import org.bouncycastle.operator.jcajce.JcaContentSignerBuilder
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.io.IOException
+import java.math.BigInteger
+import java.security.InvalidAlgorithmParameterException
+import java.security.KeyPair
+import java.security.KeyPairGenerator
+import java.security.NoSuchAlgorithmException
+import java.security.cert.CertificateException
+import java.security.cert.CertificateFactory
+import java.security.cert.X509Certificate
+import java.security.spec.ECGenParameterSpec
+import java.util.*
+
+
+class DocumentManager private constructor(private val context: Context) {
+
+    companion object {
+        private const val LOG_TAG = "DocumentManager"
+
+        @Volatile
+        private var instance: DocumentManager? = null
+
+        fun getInstance(context: Context) =
+            instance ?: synchronized(this) {
+                instance ?: DocumentManager(context).also { instance = it }
+            }
+    }
+
+    val store: IdentityCredentialStore =
+        IdentityCredentialStore.getSoftwareInstance(context)
```

Comment:
It's entirely possible some documents managed by `DocumentManager` will be in software and some in hardware. So I'm not sure it makes sense to have an instance variable for the IC store. No worries for now, we'll revisit this when we do provisioning.

Reply:
Added a TODO for now